content/control-api/curl-examples.textile

```diff
@@ -203,6 +205,330 @@ Sample response:
+
+```[sh]
+curl --location --request POST 'https://control.ably.net/v1/apps/${APP_ID/keys/${KEY_ID}/revoke' \
+--header 'Authorization: Bearer ${ACCESS_TOKEN}'
+```
+
+In this case there is only a status code returned (200), or an error code.
+
+h2(#examples-rules). Rules
+
+You can use the Control API to manage Ably "rules":/general/events. The main operations are:
+
+* List all rules for an app
+* List details of a specific rule using rule ID
+* Create a rule
+* Update a rule
+* Delete a rule
+
+h3(#examples-rules-list). List rules
+
+To list the rules for an app:
+
+```[sh]
+curl --location --request GET 'https://control.ably.net/v1/apps/${APP_ID}/rules' \
+--header 'Authorization: Bearer ${ACCESS_TOKEN}'
+```
+
+Sample response:
+
+```[json]
+[
+    {
+        "id": "oO-Tug",
+        "appId": "28GY6a",
+        "ruleType": "http",
+        "requestMode": "single",
+        "status": "disabled",
+        "source": {
+            "channelFilter": "",
+            "type": "channel.message"
+        },
+        "target": {
+            "url": "https://example.com/webhooks",
+            "signingKeyId": null,
+            "enveloped": true,
+            "format": "json",
+            "headers": [
+                {
+                    "name": "User-Agent",
+                    "value": " user-agent-name"
+                }
+            ]
+        },
+        "version": "1.0",
+        "created": 1626893758695,
+        "modified": 1626894987810,
+        "_links": {
+            "self": "https://control.ably.net/v1/apps/28GY6a/rules/oO-Tug"
+        }
+    },
+    ...
+]
+```
+
+h3(#examples-rules-list-id). List a rule by rule ID
+
+```[sh]
+curl --location --request GET 'https://control.ably.net/v1/apps/${APP_ID}/rules/${RULE_ID}' \
+--header 'Authorization: Bearer ${ACCESS_TOKEN}'
+```
+
+The rule details are returned for the specified rule.
+
+h3(#examples-rules-create). Create a rule
+
+The following example creates a rule with the following "configuration":/general/events#configure:
+
+|_. Parameter |_. Value |_. Description |
+| @ruleType@ | http | The type of rule, in this case a webhook rule |
+| @requestMode@ | single | Request mode can be single or batched. |
+| @channelFilter@ | ^my-channel.* | An optional regular expression that allows the rule to be applied to the specified channel set |
+| @type@ | channel.message | Source type. This means that the source event that will trigger this rule is any message on the channel |
+| @url@ | https://example.com/webhooks | The webhook endpoint. This is the URL triggered for the event |
+| @format@ | json | Format of encoding for the ule, in this case JSON |
+| @name@ | User-Agent | Optional header or headers, including custom headers that might be required. In this case a user agent header is specified (this needs to be set for some providers) |
+| @value@ | user-agent-name | The value of the header |
+| @enveloped@ | true | Wrap packet in metadata |
```

Comment:
Perhaps we could elaborate on what this metadata consists of? Why would we consider doing this?

Reply:
Yes, it's a bit involved, so rather than explain in the table I've added a link to our docs that describes that in detail.